sql/parser/eval.go

```diff
@@ -856,7 +842,7 @@ func (expr *ComparisonExpr) Eval(ctx EvalContext) (Datum, error) {
 	}
 
 	_, newLeft, newRight, not := foldComparisonExpr(expr.Operator, left, right)
-	d, err := expr.fn.fn(newLeft, newRight, &expr.cache)
+	d, err := expr.fn.fn(ctx, newLeft, newRight)
```

Comment:
i believe this change makes `ComparisonExpr.cache` unusued. could you remove that field?